### ISSUE TITLE
AuthorityValueService getAuthorityValueType() should match on field name, not type name

### ISSUE THREAD

**Issue by kshepherd** [DSpace/DSpace#2990]
**Describe the bug**
`AuthorityValueService.getAuthorityValueType(metadataString)` should - I think! - match the supplied string against the field of each configured value (eg "dc_contributor_author"), not the type name (eg. "Orcidv2AuthorityValue") before returning.

**Expected behaviour**
The method should treat the supplied metadata string as a field name, not a type name

**To reproduce**
Inspect the output of `getAuthorityValueType(...)` when passed a field name that is definitely configured for authority control in spring XML.

**Notes**
Method is here:
https://github.com/DSpace/DSpace/blob/737d3e677ef0c5762a6d70da7298d88fcc3cad80/dspace-api/src/main/java/org/dspace/authority/AuthorityValueServiceImpl.java#L197

This isn't actually used anywhere to my knowledge so is pretty low priority, but I noticed it when doing some authority customisation work that required me to find the class configured for a field in advance.

It is possible that I'm misunderstanding the original intent of this method - it is not commented.

**Comment by tdonohue** [comment]
@kshepherd : I'm not sure I fully understand...it looks, to me, like that `getAuthorityValueType()` method is looking for a match against the output of `AuthorityValue.getAuthorityType()` here: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/AuthorityValueServiceImpl.java#L200

That `getAuthorityType()` method returns the "type" of AuthorityValue we are looking at...see this README: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/README.md#extend-orgdspaceauthorityauthorityvalue-add-the-fields-and-implement-the-methods  So, for example, `PersonAuthorityValue.getAuthorityType()` [returns "person"](https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/authority/PersonAuthorityValue.java#L200)

So, I *think* this code is correct.  It looks like it is used by the CSV import process in a few areas...but I agree it's not well documented, so I'm not entirely sure if I understand it's behavior either.

For now, I'm going to put this in the "backlog" until determined it's a bug.  I'm not yet confident it is, but I could be wrong here.

**Comment by kshepherd** [comment]
@tdonohue I see what you're saying, perhaps I just misunderstood the usage. So the idea is that I can call authorityValueService.getAuthorityValueType("person") and it will return a PersonAuthorityValue?

For some reason I thought a field name would be passed to the method, not the type name. Perhaps it's just the 'metadataString' parameter name that confused me.

**Comment by tdonohue** [comment]
@kshepherd : Yes, based on my reading of the code, it really looks like that `String metadataString` param should be renamed to be `String type`.  In the Impl code, it's very obviously not looking up anything by a metadata string...instead, it is expecting you to call `authorityValueService.getAuthorityValueType("person")` to retrieve `PersonAuthorityValue`.

As previously mentioned though, the AuthorityValueService classes are **all** undocumented and uncommented.  So, I'm only answering based on what the code _seems to currently do_.  However, I honestly will admit, I don't know whether it's current behavior is **correct**. It's very possible you'll pointing out that it _should be doing one thing, while it's obviously doing something different._  I'm not an expert in this area of the code...and I honestly haven't had a chance to dig deeply into how/where this `getAuthorityValueType()` method is being used.

At the very least, it seems like this area of the code could use _much better_ code comments / JavaDocs.  So, if you have the time or interest to figure this stuff out, please do submit a PR to improve the JavaDocs, as I think it'll help us avoid this same sort of confusion in the long run.
